resip/stack/SipMessage.cxx

```diff
@@ -75,7 +75,9 @@ SipMessage::SipMessage(const SipMessage &from, const SipMessageOptions &opts)
       mCreatedTime(Timer::getTimeMicroSec())
 {
     init(from);
-    mOpts = opts;
+    if (opts) {
```

Comment:
Bracket on separate line.  :)

Also trying to wrap my head around why we are copying the unique_ptr.  Using a unique_ptr gives the user the impression you are handing of the data to be managed by the class, so why copy here?  If the copy is necessary, we should think about changing the constructor signature to perhaps just take in a raw pointer.

Reply:
Apologies for going quite for some time, other work stuff came up.

> Using a unique_ptr gives the user the impression you are handing of the data to be managed by the class

Not unless unique_ptr is passed by const reference (assuming no foul play by receiver of the const ref, like casting const away). 

> If the copy is necessary, we should think about changing the constructor signature to perhaps just take in a raw pointer.

I wanted to avoid using raw pointer to prevent it from being accidentally stored(copied) and outliving the unique_ptr that manages it.

When deciding on how best to pass `SipMessageOptions` to the constructor I used this [SO answer](https://stackoverflow.com/a/8114913/258825) for inspiration and reference. Let me know if reasoning above makes sense to you or if you'd like to switch to raw pointer approach in the end.

Reply:
> If a copy is needed, then the caller can copy the unique_ptr before passing it in. Thoughts?

Done

----

> Do you intend to submit a new processor or some new functionality to repro to make use of these changes publicly?

I do not have such plans at the moment.